diff -r -u alt-ergo-2.0.0/lib/reasoners/fun_sat.ml alt-ergo/lib/reasoners/fun_sat.ml
--- alt-ergo-2.0.0/lib/reasoners/fun_sat.ml	2017-11-14 18:01:38.000000000 +0100
+++ alt-ergo/lib/reasoners/fun_sat.ml	2018-05-24 16:10:10.418194590 +0200
@@ -29,6 +29,7 @@
 open Options
 open Format
 
+exception StepsLimitReached
 
 module Th = Theory.Main
 open Sig
@@ -691,8 +692,7 @@
     if steps_bound () <> -1
       && Int64.compare !steps (Int64.of_int (steps_bound ())) > 0 then
       begin
-	printf "Steps limit reached: %Ld@." !steps;
-	exit 1
+        raise StepsLimitReached
       end;
     { env with tbox = tbox; unit_tbox = utbox; inst = inst }
 
diff -r -u alt-ergo-2.0.0/lib/reasoners/fun_sat.mli alt-ergo/lib/reasoners/fun_sat.mli
--- alt-ergo-2.0.0/lib/reasoners/fun_sat.mli	2017-11-14 18:01:38.000000000 +0100
+++ alt-ergo/lib/reasoners/fun_sat.mli	2018-05-24 16:10:45.622082786 +0200
@@ -27,3 +27,5 @@
 (******************************************************************************)
 
 include Sat_solver_sig.S
+
+exception StepsLimitReached
diff -r -u alt-ergo-2.0.0/lib/util/numbers.ml alt-ergo/lib/util/numbers.ml
--- alt-ergo-2.0.0/lib/util/numbers.ml	2017-11-14 18:01:38.000000000 +0100
+++ alt-ergo/lib/util/numbers.ml	2018-05-24 16:12:31.849745671 +0200
@@ -26,7 +26,7 @@
 (*                                                                            *)
 (******************************************************************************)
 
-module MyZarith = ZarithNumbers
+module MyZarith = NumsNumbers
 module MyNums = NumsNumbers
 
 module Z = MyZarith.Z
