diff --git a/src/sat/sat_solvers.ml b/src/sat/sat_solvers.ml
index 97e384e..5213df7 100644
--- a/src/sat/sat_solvers.ml
+++ b/src/sat/sat_solvers.ml
@@ -23,6 +23,8 @@
 open Options
 open Format
 
+exception StepsLimitReached
+
 module type S = sig
   type t
 
@@ -580,8 +582,11 @@ module Dfs_sat : S = struct
       if steps_bound () <> -1
         && Int64.compare !steps (Int64.of_int (steps_bound ())) > 0 then
         begin
+          raise StepsLimitReached;
+(*
 	  printf "Steps limit reached: %Ld@." !steps;
 	  exit 1
+ *)
         end;
       { env with tbox = tbox; unit_tbox = utbox; inst = inst }
 
diff --git a/src/sat/sat_solvers.mli b/src/sat/sat_solvers.mli
index f4f9f60..4da53d1 100644
--- a/src/sat/sat_solvers.mli
+++ b/src/sat/sat_solvers.mli
@@ -22,6 +22,8 @@
 
 open Options
 
+exception StepsLimitReached
+
 module type S = sig
   type t
 
diff --git a/src/util/numbers.ml b/src/util/numbers.ml
index ebf8f95..f172a03 100644
--- a/src/util/numbers.ml
+++ b/src/util/numbers.ml
@@ -20,8 +20,8 @@
 (*   This file is distributed under the terms of the CeCILL-C licence         *)
 (******************************************************************************)
 
-module MyZarith = ZarithNumbers
-module MyNums = NumsNumbers
+module MyZarith = (*ZarithNumbers
+module MyNums =*) NumsNumbers
 
 module Z = MyZarith.Z
 
