diff --git a/Makefile.in b/Makefile.in
index e2b417f65..d381e8174 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1513,7 +1513,7 @@ ALTERGOMODS=util/config util/version util/emap util/myUnix util/myDynlink \
 	main/frontend
 
 ALTERGOCMO=$(addprefix $(ALTERGODIR)/src/, $(addsuffix .cmo,$(ALTERGOMODS)))
-TRYWHY3CMO=lib/why3/why3.cma
+TRYWHY3CMO=lib/why3/why3.cma lib/plugins/python.cmo
 TRYWHY3FILES=trywhy3.js trywhy3.html trywhy3.css \
 	README examples/ \
 	trywhy3_custom.css gen_index.sh fontawesome/css/font-awesome.min.css \
@@ -1521,6 +1521,7 @@ TRYWHY3FILES=trywhy3.js trywhy3.html trywhy3.css \
 	fontawesome/fonts/fontawesome-webfont.woff fontawesome/fonts/fontawesome-webfont.eot \
 	fontawesome/fonts/fontawesome-webfont.ttf  fontawesome/fonts/fontawesome-webfont.woff2 \
 	ace-builds/src-min-noconflict/ace.js ace-builds/src-min-noconflict/mode-why3.js \
+	ace-builds/src-min-noconflict/mode-python.js \
 	ace-builds/src-min-noconflict/theme-chrome.js $(JS_MAPS)
 
 trywhy3_package: trywhy3
diff --git a/src/trywhy3/trywhy3.html b/src/trywhy3/trywhy3.html
index b1f85157e..f19a32042 100644
--- a/src/trywhy3/trywhy3.html
+++ b/src/trywhy3/trywhy3.html
@@ -13,7 +13,7 @@
     <script type="text/javascript">
       var load_embedded_files = false;
       var editor_theme = "ace/theme/chrome";
-      var editor_mode = "ace/mode/why3";
+      var editor_mode = "ace/mode/python";
       var task_viewer_mode = "ace/mode/why3";
       var library_index = "examples/index.txt";
     </script>
diff --git a/src/trywhy3/why3_worker.ml b/src/trywhy3/why3_worker.ml
index a1e17ac6f..8221c4ff8 100644
--- a/src/trywhy3/why3_worker.ml
+++ b/src/trywhy3/why3_worker.ml
@@ -24,7 +24,7 @@ let () = log_time ("Initialising why3 worker: start ")
 
 (* Name of the pseudo file *)
 
-let temp_file_name = "/trywhy3_input.mlw"
+let temp_file_name = "/trywhy3_input.py"
 
 (* reads the config file *)
 let config : Whyconf.config = Whyconf.read_config (Some why3_conf_file)
